client/beefy/src/worker.rs

```diff
@@ -196,12 +215,12 @@ where
 	fn verify_validator_set(
 		&self,
 		block: &NumberFor<B>,
-		active: &ValidatorSet<Public>,
+		active: &ValidatorSet<AuthorityId>,
 	) -> Result<(), error::Error> {
-		let active: BTreeSet<&Public> = active.validators().iter().collect();
+		let active: BTreeSet<&AuthorityId> = active.validators().iter().collect();
 
 		let public_keys = self.key_store.public_keys()?;
-		let store: BTreeSet<&Public> = public_keys.iter().collect();
+		let store: BTreeSet<&AuthorityId> = public_keys.iter().collect();
 
 		let missing: Vec<_> = store.difference(&active).cloned().collect();
```

Comment:
shouldn't this check be the other way around?

do we want to verify that:
1. (current code) The keystore does _not_ contain keys other than the validator set.
2. All validators in the set have matching keys in the keystore.

Since forever this fn has checked `#1` but checking `#2` makes more sense to me..